### PR TITLE
Fixup serdeany_autoreg

### PR DESCRIPTION
**Pull request by addisoncrump**
Addresses some concerns we discussed about serdeany to ensure that it doesn't rely on features which aren't present downstream.

### CODE REVIEW

**Comment by addisoncrump** [comment]
@domenukk now the error is very different, seemingly unrelated (ld: file format not recognised :face_with_spiral_eyes:). Otherwise this is done.

File: libafl/src/executors/forkserver.rs
Comment (addisoncrump) [inline libafl/src/executors/forkserver.rs:1]: FYI: This file got roped into this change by accident but they are still minor bugfixes.

**Comment by domenukk** [comment]
Amazing, thx!

File: libafl/src/executors/forkserver.rs
Comment (domenukk) [reply libafl/src/executors/forkserver.rs:1]: Looks useful

**Pull request merged**